test/unittests/commands/create/create_on_server/test_ontology.py

```diff
@@ -0,0 +1,139 @@
+# mypy: disable-error-code="no-untyped-def"
+
+from http import HTTPStatus
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+import pytest
+
+from dsp_tools.commands.create.create_on_server.ontology import _create_one_ontology
+from dsp_tools.commands.create.create_on_server.ontology import _should_retry_request
+from dsp_tools.commands.create.create_on_server.ontology import create_all_ontologies
+from dsp_tools.commands.create.models.create_problems import CollectedProblems
+from dsp_tools.commands.create.models.create_problems import UploadProblem
+from dsp_tools.commands.create.models.create_problems import UploadProblemType
+from dsp_tools.commands.create.models.parsed_ontology import ParsedOntology
+from dsp_tools.commands.create.models.server_project_info import ProjectIriLookup
+from dsp_tools.utils.request_utils import ResponseCodeAndText
+from test.unittests.commands.create.constants import PROJECT_IRI
+
+ONTO_1_IRI = "http://0.0.0.0:3333/ontology/9999/onto-1/v2"
+ONTO_2_IRI = "http://0.0.0.0:3333/ontology/9999/onto-2/v2"
+
+
+@pytest.fixture
+def onto_1():
+    return ParsedOntology(
+        name="onto-1", label="Test Ontology", comment="A test ontology", classes=[], properties=[], cardinalities=[]
+    )
+
+
+@pytest.fixture
+def onto_2():
+    return ParsedOntology(
+        name="onto-2", label="Test Ontology 2", comment=None, classes=[], properties=[], cardinalities=[]
+    )
+
+
+class TestShouldRetryRequest:
+    def test_bad_request_without_matching_pattern_returns_false(self):
+        response = ResponseCodeAndText(status_code=HTTPStatus.BAD_REQUEST, text="Invalid ontology definition")
+        assert _should_retry_request(response) is False
+
+    def test_internal_server_error_returns_true(self):
+        response = ResponseCodeAndText(status_code=HTTPStatus.INTERNAL_SERVER_ERROR, text="Server error")
+        assert _should_retry_request(response) is True
```

Comment:
This causes the unit tests to take 5s longer. If you move the sleeping out of the check, you have 2 benefits:

- faster unit tests
- cleaner function (because I don't expect a boolean function to have side effects like sleeping. The only thing I expect it is to answer a question with "yes" or "no", within some milliseconds)